bee-api/bee-rest-api/src/endpoints/filters.rs

```diff
@@ -3,98 +3,12 @@
-}
-
-pub(crate) fn with_rest_api_config(
-    config: RestApiConfig,
-) -> impl Filter<Extract = (RestApiConfig,), Error = Infallible> + Clone {
-    warp::any().map(move || config.clone())
-}
-
-pub(crate) fn with_protocol_config(
-    config: ProtocolConfig,
-) -> impl Filter<Extract = (ProtocolConfig,), Error = Infallible> + Clone {
-    warp::any().map(move || config.clone())
-}
-
-pub(crate) fn with_tangle<B: StorageBackend>(
-    tangle: ResourceHandle<Tangle<B>>,
-) -> impl Filter<Extract = (ResourceHandle<Tangle<B>>,), Error = Infallible> + Clone {
-    warp::any().map(move || tangle.clone())
-}
-
-pub(crate) fn with_storage<B: StorageBackend>(
-    storage: ResourceHandle<B>,
-) -> impl Filter<Extract = (ResourceHandle<B>,), Error = Infallible> + Clone {
-    warp::any().map(move || storage.clone())
-}
-
-pub(crate) fn with_message_submitter(
-    message_submitter: mpsc::UnboundedSender<MessageSubmitterWorkerEvent>,
-) -> impl Filter<Extract = (mpsc::UnboundedSender<MessageSubmitterWorkerEvent>,), Error = Infallible> + Clone {
-    warp::any().map(move || message_submitter.clone())
-}
-
-pub(crate) fn with_peer_manager(
-    peer_manager: ResourceHandle<PeerManager>,
-) -> impl Filter<Extract = (ResourceHandle<PeerManager>,), Error = Infallible> + Clone {
-    warp::any().map(move || peer_manager.clone())
-}
-
-pub(crate) fn with_network_command_sender(
-    command_sender: ResourceHandle<NetworkCommandSender>,
-) -> impl Filter<Extract = (ResourceHandle<NetworkCommandSender>,), Error = Infallible> + Clone {
-    warp::any().map(move || command_sender.clone())
-}
-
-pub(crate) fn with_node_info(
-    node_info: ResourceHandle<NodeInfo>,
-) -> impl Filter<Extract = (ResourceHandle<NodeInfo>,), Error = Infallible> + Clone {
-    warp::any().map(move || node_info.clone())
-}
-
-pub(crate) fn with_bus(
-    bus: ResourceHandle<Bus>,
-) -> impl Filter<Extract = (ResourceHandle<Bus>,), Error = Infallible> + Clone {
-    warp::any().map(move || bus.clone())
-}
-
-pub(crate) fn with_message_requester(
-    message_requester: MessageRequesterWorker,
-) -> impl Filter<Extract = (MessageRequesterWorker,), Error = Infallible> + Clone {
-    warp::any().map(move || message_requester.clone())
-}
-
-pub(crate) fn with_requested_messages(
-    requested_messages: ResourceHandle<RequestedMessages>,
-) -> impl Filter<Extract = (ResourceHandle<RequestedMessages>,), Error = Infallible> + Clone {
-    warp::any().map(move || requested_messages.clone())
-}
-
-pub(crate) fn with_consensus_worker(
-    consensus_worker: mpsc::UnboundedSender<ConsensusWorkerCommand>,
-) -> impl Filter<Extract = (mpsc::UnboundedSender<ConsensusWorkerCommand>,), Error = Infallible> + Clone {
-    warp::any().map(move || consensus_worker.clone())
+pub(crate) fn with_args<B: StorageBackend>(
+    args: ApiArgsFullNode<B>,
+) -> impl Filter<Extract = (ApiArgsFullNode<B>,), Error = Infallible> + Clone {
+    warp::any().map(move || args.clone())
```

Comment:
do you need both `move` and `.clone()`?

Reply:
Yes, seems like

Without `move` it may outlive borrowed value `args`.
And witout `clone`: 

```
13 |     warp::any().map(move || args)
   |                 --- ^^^^^^^^----
   |                 |   |       |
   |                 |   |       closure is `FnOnce` because it moves the variable `args` out of its environment
   |                 |   this closure implements `FnOnce`, not `Fn`
   |                 the requirement to implement `Fn` derives from here
```

Reply:
So weird, I'm really looking forward to your axum PR :trollface: